cmake/OpenCVFindFrameworks.cmake

```diff
@@ -32,6 +32,8 @@ if(WITH_OPENMP)
   if(OPENMP_FOUND)
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
+    ocv_include_directories(${OpenMP_CXX_INCLUDE_DIR})
```

Comment:
OpenMP was used in the prototype implementation that I've sent you. For OpenCV, please, use cv::parallel_for_() with lambda function. Do not use OpenMP